internal/engine/podlogmanager.go

```diff
@@ -121,6 +110,29 @@ func (m *PodLogManager) diff(ctx context.Context, st store.RStore) (setup []PodL
 	return setup, teardown
 }
 
+func (m *PodLogManager) shouldWatchPodLogs(pod store.Pod) bool {
+	if pod.PodID == "" || len(pod.Containers) == 0 {
+		return false
+	}
+
+	// If an ID or name for the containers hasn't been created yet,
+	// weird things will happen when we try to store them in the `m.watches` map
+	for _, container := range pod.Containers {
```

Comment:
should we watch all the containers we CAN watch rather than bailing on the whole pod at the first un-populated container? (idk the right answer, just seems vaguely weird)

Reply:
this is an edge case that should only last for very short period of time (like sub-millisecond). It isn't worth optimizing for. better to just do the safe thing and not worry about it.

Reply:
(left a comment about this)

Reply:
sgtm!